orc8r/cloud/go/services/bootstrapper/servicers/registration/registration.go

```diff
@@ -67,12 +68,13 @@ func (r *RegistrationService) Register(c context.Context, request *protos.Regist
 	return res, nil
 }
 
-func RegisterDevice(deviceInfo protos.GatewayDeviceInfo, hwid *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) error {
+func RegisterDevice(deviceInfo *protos.GatewayDeviceInfo, hwid *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) error {
+	deviceInfoCopy := proto.Clone(deviceInfo).(*protos.GatewayDeviceInfo)
```

Comment:
Removed the copy. Guess the assumption here is that nobody will change a proto Message once it has been created?

Reply:
yea, since Go doesn't have const parameter qualifier, we'd have to rely on function description & maybe code analyzers. I think in majority of cases nobody would touch device info after the call & if in rare cases a caller would like to change it later, the caller can clone prior to call to RegisterDevice. 
RegisterDevice is not a frequently called function, I'd still prefer efficiency, but it's up to you if you want to go for safety...